src/ImageSharp/Memory/Allocators/MemoryAllocator.cs

```diff
@@ -30,16 +40,24 @@ public abstract class MemoryAllocator
     /// Creates a default instance of a <see cref="MemoryAllocator"/> optimized for the executing platform.
     /// </summary>
     /// <returns>The <see cref="MemoryAllocator"/>.</returns>
-    public static MemoryAllocator Create() =>
-        new UniformUnmanagedMemoryPoolMemoryAllocator(null);
+    public static MemoryAllocator Create() => Create(default);
 
     /// <summary>
     /// Creates the default <see cref="MemoryAllocator"/> using the provided options.
     /// </summary>
     /// <param name="options">The <see cref="MemoryAllocatorOptions"/>.</param>
     /// <returns>The <see cref="MemoryAllocator"/>.</returns>
-    public static MemoryAllocator Create(MemoryAllocatorOptions options) =>
-        new UniformUnmanagedMemoryPoolMemoryAllocator(options.MaximumPoolSizeMegabytes);
+    public static MemoryAllocator Create(MemoryAllocatorOptions options)
+    {
+        UniformUnmanagedMemoryPoolMemoryAllocator allocator = new(options.MaximumPoolSizeMegabytes);
+        if (options.AllocationLimitMegabytes.HasValue)
```

Comment:
For V4 I think we should introduce `UniformUnmanagedMemoryPoolOptions` which has the standard options as a property.

Reply:
Actually, `AllocatorOptions` is specific to `UniformUnmanagedMemoryAllocator`. I don't think anyone ever would need another allocator IRL. The design keeps it pluggable just in case + `SimpleGcMemoryAllocator` can be useful for diagnostic purposes since managed allocations are easy to track with standard .NET tools. But `SimpleGcMemoryAllocator` should not be used in production.

To secure users who still use it (mostly legacy code designed to workaround 1.* issues), I applied the default limits, but you cannot configure `SimpleGcMemoryAllocator` with `MemoryAllocatorOptions`.